sycl/test-e2e/Matrix/get_coordinate_ops_impl.hpp

```diff
@@ -0,0 +1,227 @@
+                 sub;
+
+             joint_matrix_load(
+                 sg, sub,
+                 acc.template get_multi_ptr<access::decorated::no>() +
+                     (sg_startx * SROWS * NUM_COLS) +
+                     sg_starty / sg_size * SCOLS,
+                 NUM_COLS, Layout);
+
+             ext::intel::experimental::matrix::joint_matrix_apply(
+                 sg, sub, [&](T &x, size_t row, size_t col) {
+                   sum_local_rows[row + global_idx * SROWS] += x;
+                   sum_local_cols[col + global_idy / sg_size * SCOLS] += x;
+                 });
+
+           } else {
+             joint_matrix<sub_group, T, Use, SROWS, SCOLS, Layout> sub;
+
+             joint_matrix_load(
+                 sg, sub,
+                 acc.template get_multi_ptr<access::decorated::no>() +
+                     (sg_startx * (SROWS / VF) * NUM_COLS * VF) +
+                     sg_starty / sg_size * SCOLS * VF,
+                 NUM_COLS * VF);
+
+             ext::intel::experimental::matrix::joint_matrix_apply(
+                 sg, sub, [&](T &x, size_t row, size_t col) {
+                   sum_local_rows[row + global_idx * SROWS] += x;
+                   sum_local_cols[col + global_idy / sg_size * SCOLS] += x;
+                 });
+           }
+
+           for (int i = 0; i < NUM_ROWS; i++) {
+             sum_local_rows[i] =
+                 reduce_over_group(sg, sum_local_rows[i], sycl::plus<>());
+             // only Groups leader perform the global reduction
+             if (global_idy % sg_size == 0) {
+               sycl::atomic_ref<TResult, sycl::memory_order::relaxed,
+                                sycl::memory_scope::device>
+                   aref(v_rows[i]);
+               aref.fetch_add(sum_local_rows[i]);
+             }
+           }
+
+           for (int i = 0; i < NUM_COLS; i++) {
+             sum_local_cols[i] =
+                 reduce_over_group(sg, sum_local_cols[i], sycl::plus<>());
+             // only Groups leader perform the global reduction
+             if (global_idy % sg_size == 0) {
+               sycl::atomic_ref<TResult, sycl::memory_order::relaxed,
+                                sycl::memory_scope::device>
+                   aref(v_cols[i]);
+               aref.fetch_add(sum_local_cols[i]);
+             }
+           }
+         }); // parallel for
+   }).wait();
+}
+
+template <typename T, typename TResult, size_t SROWS, size_t SCOLS, use Use,
+          layout Layout, size_t VF>
+void test_get_coord_op() {
+  constexpr size_t SCALE = 2;
+  static constexpr size_t Rows = SROWS * SCALE;
+  static constexpr size_t Cols = SCOLS * SCALE;
+
+  T M[Rows][Cols];
+  T Mvnni[Rows / VF][Cols * VF];
+  TResult sum_rows[Rows] = {0};
+  TResult sum_rows_ref[Rows] = {0};
+  TResult sum_cols[Cols] = {0};
+  TResult sum_cols_ref[Cols] = {0};
+
+  for (int i = 0; i < Rows; i++) {
+    for (int j = 0; j < Cols; j++) {
+      M[i][j] = i + j;
+    }
+  }
```

Comment:
@YixingZhang007 , we have `matrix_fill` in `common.hpp` to not duplicate such loops in each test.

Reply:
Thanks! I have changed this.